packages/langium/src/grammar/type-system/type-collector/inferred-types.ts

```diff
@@ -21,13 +21,15 @@ interface TypePart {
     children: TypePart[]
     end?: TypePart
     actionWithAssignment: boolean
+    offset?: number // used as identifier
```

Comment:
Using the offset of the cst node (which in the end just retrieves the content of the CST node anyway) seems unnecessarily complicated to me. Why not store the comment text associated with the AST node directly?